test/ct-test-srv/main.go

```diff
@@ -133,8 +133,8 @@ func (is *integrationSrv) addChainOrPre(w http.ResponseWriter, r *http.Request,
 	is.Lock()
 	for _, h := range cert.DNSNames {
 		if is.rejectHosts[h] {
-			is.Unlock()
 			is.rejected = append(is.rejected, addChainReq.Chain[0])
+			is.Unlock()
```

Comment:
I think a cleaner solution would be to factor out this logic into a `shouldReject` function that locks at the beginning, does a `defer is.Unlock()`, and then has the logic to check against rejectHosts, append to rejected, and return a boolean.

Reply:
Implemented in 9ece20d. Thanks!